tidy3d/plugins/smatrix/ports/rectangular_lumped.py

```diff
@@ -296,3 +299,230 @@ def _to_current_box(self, grid: Grid) -> Box:
+            )
+
+        # form a list of allowed terminal materials
+        ALLOWED_TERMINAL_MEDIA = (LossyMetalMedium, PECMedium)
+
+        # make sure that terminals are made of PEC or lossy metal
+        for name, terminal in (("ground", ground_terminal), ("signal", signal_terminal)):
+            if not isinstance(terminal.medium, ALLOWED_TERMINAL_MEDIA):
+                raise ValidationError(
+                    f"Invalid {name} terminal medium: expected either 'PEC' or 'LossyMetalMedium'. "
+                    f"Got: '{terminal.medium}'."
+                )
+
+        # 2D ground and signal shapes and axes
+        grounds_2d = ground_terminal.geometry.intersections_plane(x=x, y=y, z=z)
+        signals_2d = signal_terminal.geometry.intersections_plane(x=x, y=y, z=z)
+
+        _, coords_2d = Geometry.pop_axis((0, 1, 2), normal_axis)
+        lateral_axis_2d = coords_2d.index(lateral_axis)
+        voltage_axis_2d = 1 - lateral_axis_2d
+
+        # Check: structures intersects plane
+        if len(grounds_2d) == 0:
+            raise ValueError("Ground structure does not intersect Lumped Port plane.")
+        if len(signals_2d) == 0:
+            raise ValueError("Signal structure does not intersect Lumped Port plane.")
+
+        # Get 2 element list with index 0 = ground shape and index 1 = signal shape
+        shape_list = []
+
+        # loop over polygons / contours of terminals cut by a Lumped Port plane
+        for name, shapes in [("Ground", grounds_2d), ("Signal", signals_2d)]:
+            sel_shape = None
+
+            # If lateral_coord is specified, pick ground/signal shape based on intersection
+            if lateral_coord is not None:
+                # Define the infinite line at the given lateral coordinate
+                x1, x2 = (
+                    (-np.inf, np.inf) if voltage_axis_2d == 0 else (lateral_coord, lateral_coord)
+                )
+                y1, y2 = (
+                    (-np.inf, np.inf) if voltage_axis_2d == 1 else (lateral_coord, lateral_coord)
+                )
+                line = Geometry.make_shapely_box(x1, y1, x2, y2)
+
+                intersected_shapes = []
+
+                # Check shape intersects line at lateral_coord
+                for shape in shapes:
+                    if shape.intersects(line):
+                        intersected_shapes.append(shape)
+
+                if not intersected_shapes:
+                    raise ValidationError(
+                        f"{name} terminal was not detected at the specified lateral coordinate {lateral_coord}. "
+                        f"Please, make sure provided coordinate corresponds to a valid terminal location."
+                    )
+                # merge all loops of the terminal to one geometry
+                sel_shape = shapely.union_all(intersected_shapes)
+            else:
+                if len(shapes) > 1:
```

Comment:
similarly here: try unionize overlapping shapes, and throw an error if it's a multi-geometry.